### PR TITLE
Update installation docs

### LINKED ISSUE #990: Documentation:
• Improve Installation Guide

### PR DESCRIPTION
**Pull request by potipot**
Updating installation docs about mmcv-full. Updated to default torch version for icevision. 
Helps resolve #990
- [x] clarify torch and cuda version checking
- [x] add link to mmdetection installation guide
- [x] check if default torch and cuda installed by `pip install icevision` can support mmcv-full (torch 1.10.1, cu10.2)
- [x] simplify most common installation guide
- [x] consider removing conda section
- [x] include tabs for different installation methods user/dev
- [x] update environment.yml to point to master

cancelled:
- [ ] add gh action for conda build

### CODE REVIEW

**Comment by potipot** [comment]
the new mmcv installation section: image

**Comment by FraPochetti** [comment]
Shall we also remove the [Conda section](https://github.com/airctic/icevision/issues/422)?
cc @ai-fast-track

**Comment by potipot** [comment]
I'll check if it works and consider removing it

**Comment by potipot** [comment]
closes #990

File: _pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+[tool.poetry]
```
Comment (FraPochetti) [inline _pyproject.toml:1]: Have we added `poetry`?

File: environment.yml [hunk 2]
```diff
@@ -1,24 +1,19 @@
-name: ice
```
Comment (FraPochetti) [inline environment.yml:1]: When and how is this `environment.yml` file used?
To create a conda env?

File: setup.cfg [hunk 3]
```diff
@@ -13,62 +13,59 @@ license = Apache-2.0
...
+    importlib-metadata >=1;python_version<"3.8"
+    ipykernel >=4.10.1,<6
+    dataclasses ==0.6
     loguru >=0.5.3
-    importlib-metadata>=1;python_version<"3.8"
+
 
 [options.extras_require]
+inference =
+    icevision
```
Comment (FraPochetti) [inline setup.cfg:47]: Why did this change?

Comment (FraPochetti) [reply setup.cfg:47]: I am actually a bit confused by the changes in this entire file. Shall we move the discussion to Discord?

**Review by FraPochetti — APPROVED** [review]

**Pull request merged**